bc_obps/reporting/service/report_verification_service.py

```diff
@@ -45,18 +49,45 @@ def save_report_verification(version_id: int, data: ReportVerificationIn) -> Rep
             "scope_of_verification": data.scope_of_verification,
             "threats_to_independence": data.threats_to_independence,
             "verification_conclusion": data.verification_conclusion,
-            "visit_name": data.visit_name,
-            "visit_type": data.visit_type,
-            "other_facility_name": data.other_facility_name,
-            "other_facility_coordinates": data.other_facility_coordinates,
         }
         report_version = ReportVersion.objects.get(pk=version_id)
+
         # Update or create ReportVerification record
         report_verification, created = ReportVerification.objects.update_or_create(
             report_version=report_version,
             defaults=data_defaults,
         )
 
+        # Manage associated ReportVerificationVisit records
+        provided_visits = data.report_verification_visits
+
+        visit_ids_to_keep = []
+        for visit_data in provided_visits:
+            visit_defaults = {
+                "visit_type": visit_data.visit_type,
+                "is_other_visit": visit_data.is_other_visit,
+                "visit_coordinates": visit_data.visit_coordinates,
+            }
+
+            visit, visit_created = ReportVerificationVisit.objects.update_or_create(
+                report_verification=report_verification,
+                visit_name=visit_data.visit_name,
+                defaults=visit_defaults,
+            )
+
+            visit_ids_to_keep.append(visit.id)
+
+        # Delete any visits not included in the provided data
+        ReportVerificationVisit.objects.filter(report_verification=report_verification).exclude(
+            id__in=visit_ids_to_keep
+        ).delete()
+
+        # Fetch ReportVerificationVisit instances corresponding to the IDs
+        visit_instances_to_keep = ReportVerificationVisit.objects.filter(id__in=visit_ids_to_keep)
+
+        # Update report_verification_visits
+        report_verification.report_verification_visits.set(visit_instances_to_keep)
+
```

Comment:
I don't think this is necessary, all the `ReportVerificationVisit` models have been created with the right `report_verification` field set, and this is not an m2m relationship.
We can remove these few lines